src/MonthSelect.php

```diff
@@ -4,24 +4,78 @@
 
 namespace Laminas\Filter;
 
+use function filter_var;
+use function is_array;
+use function is_numeric;
+use function sprintf;
+
+use const FILTER_VALIDATE_INT;
+
 /**
  * @psalm-type Options = array{
  *     null_on_empty?: bool,
  *     null_on_all_empty?: bool,
- *     ...
  * }
- * @psalm-type InputArray = array{
- *       year: numeric,
- *       month: numeric,
- *  }
- * @template TOptions of Options
- * @template-extends AbstractDateDropdown<TOptions, InputArray>
+ * @implements FilterInterface<string|null>
  */
-final class MonthSelect extends AbstractDateDropdown
+final class MonthSelect implements FilterInterface
 {
-    /**
-     * Year-Month
-     */
-    protected string $format      = '%2$s-%1$s';
-    protected int $expectedInputs = 2;
+    private readonly bool $returnNullIfAnyFieldEmpty;
+    private readonly bool $returnNullIfAllFieldsEmpty;
+
+    /** @param Options $options */
+    public function __construct(array $options = [])
+    {
+        $this->returnNullIfAnyFieldEmpty  = $options['null_on_empty'] ?? false;
+        $this->returnNullIfAllFieldsEmpty = $options['null_on_all_empty'] ?? false;
+    }
+
+    public function __invoke(mixed $value): mixed
+    {
+        return $this->filter($value);
+    }
+
+    public function filter(mixed $value): mixed
+    {
+        if (! is_array($value)) {
+            return $value;
+        }
+
+        /** @var mixed $month */
+        $month = $value['month'] ?? null;
+        /** @var mixed $year */
+        $year = $value['year'] ?? null;
```

Comment:
```suggestion
        $month = $value['month'] ?? null;
        /** @var mixed $month */
        $month = $month === '' ? null : $month;
        $year  = $value['year'] ?? null;
        /** @var mixed $year */
        $year = $year === '' ? null : $year;
```
Sorry to be a pain, but we should probably coalesce the empty string to null for both `$year` and `$month` before comparison too, and add those to the tests because if you've set these options to true, you probably want `null` when `$input === ['month' => '', 'year' => '']`